formats/xlsx.js

```diff
@@ -44,6 +44,8 @@ Stream.prototype._flush = function (done) {
     workbook.SheetNames.forEach(function (name) {
         var sheet = workbook.Sheets[name]
         var data = []
+        // Find the first row of the table
+        var headerRow = Object.keys(sheet)[1].match(regexp)[2][0]
```

Comment:
Since `Object.keys` doesn't guarantee order of keys, I think you should sort the keys (will probably require a custom comparator function to correct sort cases like `['B9', 'A49', 'A401']` - classic lexical sorting pitfall) otherwise the key at `[1]` might not be the row you want

Further, you're making a lot of direct array access which might not exist (if the regex fails for example for some reason)

Lastly, I know its not your naming, but calling a regex variable `regex` isn't very informative. One on line 52 did I understand whats the actual purpose of `regex` -  I think its worth renaming